proxy/http/HttpConfig.cc

```diff
@@ -900,6 +900,59 @@ register_stat_callbacks()
                      (int)http_sm_finish_time_stat, RecRawStatSyncSum);
 }
 
+static bool
+set_negative_caching_list(const char *name, RecDataT dtype, RecData data, HttpConfigParams *c, bool update)
+{
+  bool ret = false;
+  HttpStatusBitset set;
+  // values from proxy.config.http.negative_caching_list
+  if (0 == strcasecmp("proxy.config.http.negative_caching_list", name) && RECD_STRING == dtype && data.rec_string) {
+    // parse the list of status code
+    ts::TextView status_list(data.rec_string, strlen(data.rec_string));
+    while (!status_list.empty()) {
+      auto is_sep{[](char c) { return isspace(c) || ',' == c || ';' == c; }};
+      ts::TextView span, token{status_list.ltrim_if(is_sep).take_prefix_if(is_sep)};
```

Comment:
You need to either move `is_sep` and the `ltrim_if` up, or handle the possibility of an empty `token` if there are trailing separators. I'd favor the former - use `while (!status_list.ltrim_if(is_sep).empty()) {` and move the `is_sep` definition to just before the `while`.

Reply:
right